src/lib/services/segment-service.ts

```diff
@@ -300,11 +314,13 @@ export class SegmentService implements ISegmentService {
         id: number,
         segment: Omit<ISegment, 'id'>,
     ): Promise<void> {
-        const strategies =
-            await this.featureStrategiesStore.getStrategiesBySegment(id);
+        const { strategies, changeRequestStrategies } =
+            await this.getAllStrategies(id);
 
         const projectsUsed = new Set(
-            strategies.map((strategy) => strategy.projectId),
+            [strategies, changeRequestStrategies].flatMap((strats) =>
+                strats.map((strategy) => strategy.projectId),
+            ),
```

Comment:
This prevents you from turning a global segment to a project segment if it is used in CRs in different projects.